## Microbiomes and Livestock
 _How do different types of interactions with livestock affect the microbiome diversity of young children?_
#### Introduction

The human microbiome has been a recently trendy topic of study in the world of biology, as researchers try to untangle the relationships between the human microbiome and different aspects of human health. Studies have suggested connections between the microbiome and everything from digestion to immune response (Shreiner et al, 2015). Additionally, the makeup of the microbiome is believed to be influenced by many factors including diet, age, and environment (Xu & Knight, 2015). 

The main objectives of this independent investigation are to consider how human interactions with common livestock animals, specifically cows and chickens, can affect the diversity of microbes found in the human gut microbiome. If interactions with animals can alter the diversity of the microbiome, then further questions are raised about animal interactions and their effects on human health as a whole.
#### Methods
##### Available Data
The data that I will be using comes from the paper [_"Microbiome sharing between children, livestock and household surfaces in western Kenya"_](https://doi.org/10.1371/journal.pone.0171017) by Emily Mosites et al. The goal of the original investigation was to compare the gut microbiomes of children, livestock, and living areas in rural Kenya. The study by Mosites et al. aimed to determine if interactions with livestock led to higher similarities between the microbiomes of children and animals. Additionally, they investigated what factors, such as wealth and antibiotic use, contributed to microbiome diversity in children. 

The original study collected stool samples from children and cows, cloacal swabs from chickens, and swabs of living spaces from 158 households in western Kenya. DNA was extracted from the samples and PCR amplified in order to identify the taxonomic makeup of the microorganisms in the microbiome of each sample. Microbiome diversity was measured in Operational Taxonomic Units (OTU), a metric of the number of different species or genera found in a sample. OTU counts were filtered to remove samples with a low number of reads, OTUs that were found in only one sample, and OTUs that could not be identified in a database. The diversity and makeup of the microbiomes was compared between species and households to determine if microbiomes are affected by environment and cross-species interactions. Mosites et al. also collected data on the different types of interactions that children had with livestock, including feeding, herding, milking, and cleaning up after the animals. 

The study by Mosites et al. reported findings on the different taxonomic groups found in the microbiomes of different species and then compared the similarities of samples from the same household vs. those from different households. They also considered how different factors including child age, livestock feeding, and household wealth score affected the diversity of children's microbiomes. The authors found that reported child interactions with livestock were not significantly associated with similarities in microbiome makeup between children and livestock. Mosites et al. did look into how different factors may affect child microbiome diversity, and they reported that feeding livestock was found to have a significant effect on microbiome diversity.   The authors do not mention any of the other livestock interactions, possibly because they did not have a significant effect on diversity, but it is unclear and investigating effects on diversity was not their primary analysis. 

The original data is posted on [Dryad](https://doi.org/10.5061/dryad.f7tp6) as a collection of Excel sheets. There is also a README file, but it only contains contact info for the original authors. The metadata is mostly on one of the Excel sheets, which is a full ["Data Dictionary"](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/Mosites_Microbiome_Sharing_Kenya_20170119.xlsx) with explanations of all of the variables that were used in the original study. In the Dryad data collection, there is one sheet with interview data, such as how the different members of the household interact, and one sheet with biome data, including taxonomic diversity. I saved both Excel sheets as CSV files, which were 13 KB with 144 lines of interview data and 47 KB with 410 lines of biome data. For the most part, the data is well organized and clear, but there were some issues that arose from the use of **()** and **%** symbols within entries that prevented columns from being imported into R as numeric data. There were also some missing data entries in both the biome and interview data that were removed before my independent analysis. There was no code from the original study included in the Dryad repository.

##### Independent Investigation Methods
_Goal:_ Investigate how different types of interactions with livestock affect the gut microbiome diversity of children

In order to use the Mostites et al. data for my independent analysis, I first saved the available Excel sheets as .csv files so that I could edit them in the terminal and import them to R for analysis. I tried to import the biome and interview files straight to R without doing any data clean-up, but the use of **()** and **%** symbols in some of the biome data entries prevented some necessary columns from being imported as numeric data. To solve this, I used [a pipeline](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/cleaning_script.sh) of two `sed`commands to remove the unnecessary characters and save a [cleaned-up version of the biome data csv](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/cleanedBiomes.csv). 

After cleaning up the biome data, I was able to import both csv files into R with all of the necessary columns as integers. The study by Mosites et al. included a wide range of variables, and I did not need them all to answer my investigation question. I used the `select()` function of the `dplyr` package to efficiently subset the columns that I needed from both the biome and interview datasets. I also used the `filter()` function to remove all non-human biome entries since I looked exclusively at the microbiome diversity of the children sampled in the Mosites et al. study. 

As final preparation for visualization and analysis, I combined the interview data and the biome data using a full join. This allowed me to keep all of the rows for both data frames, but have them combined by shared household IDs. I also removed all rows that did not have both biome and interview data so that `NAs` were not included in the final visualizations and incomplete data was not included in calculations. This was relatively easy to do because there were no individual entries missing, only whole biome or interview sets. Households either had all of the interview data or none of it, and the same was true for the biome data. This meant I could remove rows that had an `NA` in just one column of the interview data or just one column of the biome data. I also made an additional subset of the combined data that included all children that had no interaction with livestock. This was used as a control that is included in all of the final visualizations.

To visualize the interaction data, I used `ggplot2`, which required the interaction columns to be factors rather than integers. The `as.factor()` function was not working with column indexing inside of a for loop, so I created an additional loop to make all of the interaction columns factors before graphing. To produce a series of boxplots, one for each different type of livestock interaction compared to the unfiltered OTU counts, I wrote a for loop of ggplots that saved each plot as a variable. I also included optional lines that will save each plot as an image in the working directory or print each one in the R Studio console. The final visualization for my analysis is the series of boxplots for the different livestock interactions all combined into one image, which I did using the `gridExtra` package. 

In order to make sense of the boxplots statistically, I ran a series of t-tests to determine if the mean diversity of children with a certain interaction type was significantly different from the diversity of children without that interaction type. I ran one test for each interaction type using a for loop and a conditional which prints out _"Difference is significant"_ for interactions with p-values below 0.05. I would have liked to do an additional T-test or an ANOVA to compare the different interactions to the controls, but I did not think of this until later in the project, so I did not have time to get that done. 

After viewing the results from the t-tests, I decided to rerun the same visualizations and statistical tests on the filtered OTU data that was provided by Mosites et al. The filtered data does not include samples with a low number of reads, OTUs that were found in only one sample, and OTUs that could not be identified in a database. I used the same code and series of steps for this secondary analysis, but specified the filtered OTU column in the biome data. All R code that I wrote can be found [here](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/animal_interactions.R).

#### Results and Conclusions
The boxplots that resulted from my code were useful for visually comparing how different livestock interactions relate to microbiome diversity. 

**Unfiltered OTU Count Plots**
![unfilteredplots](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/FinalUnfilteredPlots.png?raw=true)

_The unfiltered OTU count plots above show each different livestock interaction and the corresponding OTU Counts of children who interacted with livestock in this way. The control is the same for each plot and represents children who did not have any interaction with livestock (n = 9). There appears to be strong differences in diversity for multiple different interaction types, including caring and marketing, but also a large number of drastic outliers in many of the plots._

**Filtered OTU Count Plots**
![filteredplots](https://github.com/msbrasher/CompBioLabsHw/blob/main/FinalProject/FinalFilteredPlots.png?raw=true)

_There is a noticeable difference between the filtered and unfiltered OTU plots, suggesting that biome data that was low-quality or included unidentifiable OTUs had a significant effect on the layout of the data. There is much less visually discernable difference in diversity for nearly all of the interaction types. It appears that many outliers were also removed through filtering. Additionally, all interactions seem to have a median OTU count around 1500, as compared to 4000 for the unfiltered OTU counts._

The t-tests that were run for each interaction type for both filtered and unfiltered data determined that the large majority of interactions did not have a statistically significant affect on microbiome diversity. Using the unfiltered data, milking was determined to have a significant effect (p-value = .00379), but the same did not hold true when testing the filtered data. For the filtered data, feeding was determined to have a significant effect on diversity (p-value = .02007). This supports what Mosites et al. found in their investigation using the same data, as they reported that feeding livestock had a significant effect on microbiome diversity along with household wealth score and child age.  

While these results do suggest that some interactions with livestock can increase the microbiome diversity of children, there are still many remaining questions about how human health may be impacted. One major question is why feeding livestock seems to affect microbiome diversity but activities like milking and cleaning do not. This type of analysis could also be expanded to consider different species of animals, different contexts of interactions such as in a professional animal care setting, and considering adult microbiomes which are generally more established than those of children. Ultimately, there is no doubt that I learned a lot from this investigation through challenging and expanding my computational skills, but, as is typical with scientific investigations, far more questions were raised than were answered. 
#### References

Mosites, Emily et al. (2018), Data from: Microbiome sharing between children, livestock and household surfaces in western Kenya, Dryad, Dataset, [https://doi.org/10.5061/dryad.f7tp6](https://doi.org/10.5061/dryad.f7tp6)

Mosites et al., “Microbiome Sharing between Children, Livestock and Household Surfaces in Western Kenya," [https://doi.org/10.1371/journal.pone.0171017](https://doi.org/10.1371/journal.pone.0171017)

Shreiner, A. B., Kao, J. Y., & Young, V. B. (2015). The gut microbiome in health and in disease. _Current opinion in gastroenterology_, _31_(1), 69–75. [https://doi.org/10.1097/MOG.0000000000000139](https://doi.org/10.1097/MOG.0000000000000139) 

Xu, Z., & Knight, R. (2015). Dietary effects on human gut microbiome diversity. _British Journal of Nutrition,_  _113_(S1), S1-S5. [doi:10.1017/S0007114514004127](https://doi.org/10.1017/S0007114514004127) 

> Written with [StackEdit](https://stackedit.io/).
